src/bootstrap/compile.rs

```diff
@@ -330,6 +330,8 @@ impl Step for StartupObjects {
                                     obj);
             builder.copy(&src, &sysroot_dir.join(obj));
         }
+        let libgcc = compiler_libgcc(builder, builder.cc(target));
+        builder.copy(&libgcc, &sysroot_dir.join("libgcc.a"));
```

Comment:
This seems like it may be indicative of a bug? This isn't necessary on other platforms and may indicate a misconfigured toolchain?

Reply:
Not necessarily:
```
chouquette@nibbler /opt/gcc-i686/bin/i686-w64-mingw32-gcc -print-libgcc-file-name         
/opt/gcc-i686/lib/gcc/i686-w64-mingw32/6.4.0/libgcc.a
chouquette@nibbler /opt/llvm-toolchains/bin/i686-w64-mingw32-gcc -print-libgcc-file-name  
/opt/llvm-toolchains/lib/clang/9.0.0/lib/windows/libclang_rt.builtins-i386.a
chouquette@nibbler /opt/llvm-toolchains/bin/x86_64-w64-mingw32-gcc -print-libgcc-file-name
/opt/llvm-toolchains/lib/clang/9.0.0/lib/windows/libclang_rt.builtins-x86_64.a
```
rust used to use libgcc.a unconditionally, but that doesn't work with llvm/clang since the name of the library is different, and will change depending on the platform.

Or it might be possible to get the absolute path to that library and provide it as a post_link_object directly, AFAIR I went this way to try and stick to the existing assumptions.

Unless your point is that llvm/clang should provide a compatibility libgcc.a?

Reply:
Hm no my point is that this was never needed before now so it should continue to not be needed for all existing platforms. If this is uwp-specific then it seems like it's likely specific to the uwp mingw toolchain, and it needs to be documented as to why it's needed for uwp and only affect uwp

Reply:
As far as I can see, the folder containing the various `libclang_rt...a` are not present by default. Those can't be known in advance as they are heavily dependent on the clang installation directory.
The best option I can think of is to have a clang only code path, that would copy the file in the directory used for `crt.o` & `dllcrt2.o`, and tweak the spec files so they add `libvlc_rt.builtin-<arch>.a` to their `post_link_object` (if you'd prefer I can make that a UWP only path for now.)

This way the regular gcc-mingw builds are unaffected, and the llvm-mingw builds should work, but I don't see a way around copying the clang_rt libraries out of the installation path.

Reply:
Er well sorry but to get to the point this needs to be removed. This, along with `libgcc.a` below in the target definition, should not be necessary. They can be added in a build script directive if really need, otherwise requiring these here and there is likely a bug in the build process that shouldn't be fixed like this.

Reply:
I agree that it should be removed, but to be fair, I'm not sure I know how.
Removing `-nodefaultlibs` would be a way, but I'm pretty sure it's there for a reason, so that seems like a no go.
I can definitely do something without touching the `windows_base`, but it will prevent the toolchain to be built with llvm.
If the change needs to work for both `pc-windows-gnu` and `uwp-windows-gnu`, then I really don't know how to point rust's build system to the correct `libgcc.a` or `libclang_rt...a`

Reply:
I think it's fine to just remove `-nodefaultlibs` for the UWP target or something like that. Basically this fix should be UWP specific and shouldn't accidentally touch other platforms.